agent/php_execute.c

```diff
@@ -491,6 +491,10 @@ static nr_library_table_t libraries[] = {
 
     {"MongoDB", NR_PSTR("mongodb/src/client.php"), nr_mongodb_enable},
 
+    /* php-amqplib RabbitMQ >= 3.7 */
+    {"php-amqplib", NR_PSTR("phpamqplib/connection/abstractconnection.php"),
+     nr_php_amqplib_enable},
```

Comment:
Since `phpamqplib/connection/abstractconnection.php` file also exists in version 2.x  of php-amqplib (starting with version [v2.0.2](https://github.com/php-amqplib/php-amqplib/blob/v2.0.2/PhpAmqpLib/Connection/AbstractConnection.php)), this code will enable agent's instrumentation of php-amqplib in projects that use version 2 of php-amqplib. Is this safe?

Reply:
Preliminary testing shows it basically works fine and creates the message segments, but we aren't fully testing it.

Would your vote be to allow it or to force a check and not instrument if it's a version below 3.7.

Reply:
Are the signatures of `PhpAmqpLib\Channel\AMQPChannel::basic_publish` and `PhpAmqpLib\Channel\AMQPChannel::basic_get` methods the same between 2.x and 3.x?

Reply:
Yes

Reply:
If the method signatures are the same, then I would leave things as they are.